Modify Search and Rescue behavior

rodan123 opened the pull request:
Turning in a search and rescue item should pass the same variables as a commodity sell to VoiceAttack.
Changed code to send item to VA as _EDDI search and rescue commodity_.
 Standalone script was only using the rare flag in addition to the name, not worth denying the item name to VA for that. If we really need the rare flag, it should be sent as a separate variable.
Modified standalone script, removed _Humanise()_ from reward, it was making the sentence nonsensical at times.

Tkael commented:
Why not make both the raw name and the object available? The name could be sent to VA under 'EDDI search and rescue name' while still preserving the information that is made available from the current object.

Please provide an example - how was Humanise() making the reward nonsensical?

Tkael commented:
Firstly, for backwards compatibility. Changing the variable from event.commodity.name to event.commodity is an easily avoidable script breaking change for everyone using the current script.

Secondly, as a philosophy, we generally try to make as much information available at possible, even if it isn't used by the default script (power users may find uses we didn't anticipate)

Regarding Humanise(), I'd like to keep that in as it generally improves the flow of the dialog, but also create a second pull request with your proposed change to that function.

rodan123 commented:
You make a good point about not breaking scripts.  I have reworked the code to provide an additional variable and leave the commodity object as it is. I named the variable for VA as _EDDI search and rescue commodityname_ , a little wordy, but that way it closely matches the commodity object's _commodity.name_ which it mirrors. Will test and post up. I reverted all the changes to the S&R cottle script, except for the single typo correction, so it won't conflict with any changes you make. Let me know if you want me to do the new pull request for altering the "Humanize" function.

Tkael commented:
Yes please do. I'd like to give the team a chance to weigh in on the proposed change to Humanise() and a separate PR is a good way to manage that.

Tkael commented on speechresponder/eddi.json:
```diff
@@ -1275,7 +1275,7 @@
       "priority": 3,
       "responder": true,
-      "script": "{event.amount} \r\n{if event.commodity.rare: rare}\r\n\r\n{if event.commodity.name = \"Black Box\":\r\n   {if event.amount > 1:\r\n      {OneOf(\"flight records\", \"black boxes\")}\r\n   |else: \r\n      {OneOf(\"flight record\", \"black box\")}\r\n   }\r\n\r\n|elif event.commodity.name = \"Occupied CryoPod\":\r\n   {OneOf(\"damaged cryo pod\", \"occupied cryo pod\")}{if event.a ...
+      "script": "{event.amount} \r\n{if event.commodity.rare: rare}\r\n\r\n{if event.commodity.name = \"Black Box\":\r\n   {if event.amount > 1:\r\n      {OneOf(\"flight records\", \"black boxes\")}\r\n   |else: \r\n      {OneOf(\"flight record\", \"black box\")}\r\n   }\r\n\r\n|elif event.commodity.name = \"Occupied CryoPod\":\r\n   {OneOf(\"damaged cryo pod\", \"occupied cryo pod\")}{if event.a ...
```
Please revert this change to prevent a merge conflict with PR #263 (since both would otherwise modify the same file). I've incorporated the correction into that PR.

rodan123 commented:
OK - Think it is good to go, now.

Tkael commented:
Hmm. It jumped from two files modified to four, and the diff is now including additional items.
Would it be possible to resubmit a cleaned version of this PR with only the changes to Events/SearchAndRescueEvent.cs and a description of the change in EDDI/ChangeLog.md?

rodan123 commented:
Sure no problem. Closing this pull request.
